aten/src/ATen/native/quantized/cuda/fake_quantize_core.cu

```diff
@@ -25,17 +27,22 @@ void fake_quantize_slice_kernel_cuda(
     int64_t quant_min,
     int64_t quant_max) {
   float inv_scale = 1.0f / scale;
-  at::cuda::CUDA_tensor_apply2<float, float>(
-      input, output, [=] __device__(const float& input_val, float& result_val) {
-        result_val = (fminf(
-                          quant_max,
-                          fmaxf(
-                              quant_min,
-                              static_cast<int64_t>(std::nearbyint(
-                                  input_val * inv_scale + zero_point)))) -
-                      zero_point) *
-            scale;
-      });
+  auto iter = TensorIterator();
+  iter.dont_compute_common_dtype();
+  iter.add_output(output);
+  iter.add_input(input);
+  iter.build();
+  gpu_kernel(iter,
+    [=] GPU_LAMBDA (float input_val) -> float {
+      return (fminf(
```

Comment:
Does integer min/max function work here?

Reply:
Hmmm, int32 could have range issue, int64 version should work, but on GPU, float32 is faster than int64
https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#arithmetic-instructions

Reply:
could you benchmark this? since using float is conversion + comparison and using int64_t is just comparison

Reply:
sure

Reply:
@jerryzh168 I didn't see the difference in the performance of these two cases

Reply:
OK, thanks!